### ISSUE TITLE
Transition To Workbench in May

### ISSUE THREAD

**Issue by zkamvar** [swcarpentry/python-novice-gapminder#632]
@swcarpentry/python-novice-gapminder-maintainers

As I hope you are already aware, we are rolling out the new lesson infrastructure, [The Carpentries Workbench](https://carpentries.github.io/workbench/), across all of The Carpentries official lessons in early May 2023. This means that all Data Carpentry, Library Carpentry, and Software Carpentry lesson repositories will be modified to adopt the new infrastructure in the coming days.

You can follow the transition of this lesson repository at https://github.com/carpentries/lesson-transition/issues/77.

The transition has already taken place for several lessons, and so far the process has been running quite smoothly. You should see the transition take place with minimal disruption, but there are a few things that it is important for Maintainers to be aware of. 

Here is what you can expect to happen next:

1. Any open pull requests on the repository will be closed with an automated message.
2. The repository will be set to read-only mode for a brief period while the transition occurs.
3. The new repository structure and lesson site layout will then be applied.
4. To avoid anyone accidentally pushing the old commit history back to the repository, after the transition Maintainers will need to delete and replace any existing forks and local clones they have of the lesson repository, and confirm that they have done so by replying to this issue.

I will reply here before and after the transition has taken place. If you have any questions in the meantime, please reach out to the Curriculum Team by tagging us here, e.g. @swcarpentry/core-team-curriculum.

If you would like to read more about the new lesson infrastructure and the modified repository structure you can expect post-transition, I recommend [the Infrastructure episode of the Maintainer Onboarding curriculum](https://carpentries.github.io/maintainer-onboarding/05-infrastructure.html) and [the Workbench Transition Guide](https://carpentries.github.io/workbench/transition-guide.html), which includes a side-by-side comparison of various elements of the old and new infrastructures.

**Comment by zkamvar** [comment]
Hello maintainers, one thing I notice in [The previewe version of the Lesson Design page](https://fishtree-attempt.github.io/python-novice-gapminder/instructor/design.html#stage-3-learning-plan) (from [_extras/design.md](https://github.com/swcarpentry/python-novice-gapminder/blob/gh-pages/_extras/design.md)) is that there are episode links that are no longer valid (see the missing files listed in https://github.com/carpentries/lesson-transition/issues/77). I can go in and fix these, but I'm wondering if I should just leave these links alone given that the file is out of sync with the lesson and has not been updated in 3 years.

**Comment by zkamvar** [comment]
(Note: I _just_ rebuilt the preview, so the link may not be active for another couple of minutes)

**Comment by alee** [comment]
Thanks for catching that @zkamvar! I'd vote to leave those alone and we can address them in a separate issue / PR after the workbench transition, don't want to add to your burdens.

**Comment by zkamvar** [comment]
We are all set to transition this lesson tomorrow. I have fixed all the issues that were identified and you can see the preview here: https://fishtree-attempt.github.io/python-novice-gapminder/

**Comment by alee** [comment]
The landing page has this include still sitting around, I assume it should be removed?

```
{% include gh_variables.html %}
```

![image](https://user-images.githubusercontent.com/22534/235581560-b45f1188-5a07-45fa-90b8-8a685450d73d.png)

**Comment by zkamvar** [comment]
This lesson will be converted to use [The Carpentries Workbench][workbench]
To prevent accidental reversion of the changes, we are temporarily revoking
write access for all collaborators on this lesson:

 - [ ] @alee (push)
 - [ ] @richmccue (push)
 - [ ] @coatless (push)
 - [ ] @annajiat (push)
 - [ ] @souravsingh (push)
 - [ ] @vahtras (push)
 - [ ] @martinosorb (push)
 - [ ] @mxiang1 (push)
 - [ ] @Gina123-P (push)

If you no longer wish to have write access to this repository, you do not
need to do anything further.

1. What you can expect from the transition 📹: https://carpentries.github.io/workbench/beta-phase.html#beta
2. How to update your local clone 💻: https://carpentries.github.io/workbench/beta-phase.html#updating-clone
3. How to update (delete) your fork (if you have one) 📹: https://carpentries.github.io/workbench/faq.html#update-fork-from-styles

If you wish to regain write access, please re-clone the repository on your machine and
then comment here with `I am ready for write access :rocket:` and the
admin maintainer of this repository will restore your permissions.

If you have any questions, please reply here and tag @zkamvar

[workbench]: https://carpentries.github.io/workbench

**Comment by zkamvar** [comment]
The deed is done. The infrastructure takes a few minutes to bootstrap and cache the packages for the lesson build. Once the build is done, I will switch github pages to deploy from the `gh-pages` branch and you will have your workbench lesson. 

Thank you all for your enthusiasm and your patience!

**Comment by alee** [comment]
Thanks @zkamvar! Looks great. I've re-forked the repo and reset my local git repo to upstream/main 

Are there any plans for Docker support to render content locally? I'd be happy to contribute to https://github.com/carpentries/workbench/issues/39 if help is still needed there, I'm pretty comfortable with Docker / docker compose / etc

I am ready for write access :rocket:

**Comment by vahtras** [comment]
I am ready for write access :rocket:

**Comment by zkamvar** [comment]
@vahtras you now have write access!

**Comment by souravsingh** [comment]
I am ready for write access :rocket:

**Comment by zkamvar** [comment]
@souravsingh access granted!

**Comment by martinosorb** [comment]
I am ready for write access :rocket: 
Thanks

**Comment by zkamvar** [comment]
@martinosorb, access granted! 

All maintainers on this lesson now have access!
